irods_consortium_continuous_integration_build_hook.py

```diff
@@ -9,70 +10,53 @@
 import irods_python_ci_utilities
 
 
-def install_cmake_and_add_to_front_of_path():
-    irods_python_ci_utilities.install_os_packages(['irods-externals-cmake3.11.4-0'])
-    os.environ['PATH'] = '/opt/irods-externals/cmake3.11.4-0/bin' + os.pathsep + os.environ['PATH']
-
-
-def get_build_prerequisites_all():
-    return ['irods-externals-avropre190cpp17-0',
-            'irods-externals-boost1.67.0-0',
-            'irods-externals-clang-runtime6.0-0',
-            'irods-externals-clang6.0-0',
-            'irods-externals-cppzmq4.2.3-0',
-            'irods-externals-jansson2.7-0',
-            'irods-externals-libarchive3.3.2-1',
-            'irods-externals-qpid-with-proton0.34-1',
-            'irods-externals-zeromq4-14.1.6-0']
-
-
-def get_build_prerequisites_apt():
-    return ['libcurl4-gnutls-dev', 'make', 'libssl-dev', 'gcc'] + get_build_prerequisites_all()
-
-
-def get_build_prerequisites_yum():
-    return ['curl-devel', 'openssl-devel'] + get_build_prerequisites_all()
-
-
-def get_build_prerequisites():
-    dispatch_map = {
-        'Ubuntu': get_build_prerequisites_apt,
-        'Centos': get_build_prerequisites_yum,
-        'Centos linux': get_build_prerequisites_yum,
-        'Opensuse ': get_build_prerequisites_yum,
-    }
-    try:
-        return dispatch_map[irods_python_ci_utilities.get_distribution()]()
-    except KeyError:
-        irods_python_ci_utilities.raise_not_implemented_for_distribution()
-
-
-def install_build_prerequisites_apt():
-    if irods_python_ci_utilities.get_distribution() == 'Ubuntu': # cmake from externals requires newer libstdc++ on ub12
-        if irods_python_ci_utilities.get_distribution_version_major() == '12':
-            irods_python_ci_utilities.install_os_packages(['python-software-properties'])
-            irods_python_ci_utilities.subprocess_get_output(['sudo', 'add-apt-repository', '-y', 'ppa:ubuntu-toolchain-r/test'], check_rc=True)
-            irods_python_ci_utilities.install_os_packages(['libstdc++6'])
-    irods_python_ci_utilities.install_os_packages(get_build_prerequisites())
-
-
-def install_build_prerequisites_yum():
-    irods_python_ci_utilities.install_os_packages(get_build_prerequisites())
-
-
-def install_build_prerequisites():
+def add_cmake_to_front_of_path():
+    cmake_path = '/opt/irods-externals/cmake3.11.4-0/bin'
+    os.environ['PATH'] = os.pathsep.join([cmake_path, os.environ['PATH']])
+
+def install_building_dependencies(externals_directory):
+    externals_list = ['irods-externals-cmake3.11.4-0',
+                      'irods-externals-avro1.9.0-0',
+                      'irods-externals-boost1.67.0-0',
+                      'irods-externals-clang-runtime6.0-0',
+                      'irods-externals-clang6.0-0',
+                      'irods-externals-cppzmq4.2.3-0',
+                      'irods-externals-json3.7.3-0',
+                      'irods-externals-jansson2.7-0',
```

Comment:
we still need jansson?

Reply:
Needs testing.

Reply:
Just tested it we require jansson it is one of the dependencies

Reply:
sure, it's listed/used, but the question is whether we need it.    the json3.7.3 library should do everything we need.

Reply:
created #42